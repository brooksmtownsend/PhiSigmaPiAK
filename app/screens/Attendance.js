import React, { Component } from 'react';
import { View, Text, WebView, TextInput, Button, TouchableOpacity, Keyboard, ActivityIndicator, SafeAreaView, Image } from 'react-native';

/**
 * Attendance is our component for PSP event sign-in
 * State is composed of:
 * uriPath: Attendance code and TinyURL postfix
 * shouldRenderWebView: When true will render WebView, starts false and changes upon clicking the button
 */
export default class Attendance extends Component {

  constructor() {
    super();
    this.state = {
      uriPath: '',
      shouldRenderWebView: false,
    };
  }

  // Callback function that is called when the button is pressed, or TextInput is done editing.
  renderWebView() {
    Keyboard.dismiss();
    this.setState({shouldRenderWebView: true});
  }

  // Function returns an activity indicator for idle loading
  renderLoadingView() {
    return (
      <ActivityIndicator color="#0D6275" size="large" style={{
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 99999
    }}/>
    );
  }

  render() {
    return (
      <SafeAreaView style={{flex:1}}>
        <View style={{flex:1}}>
          <Text style={{alignSelf: 'center', fontSize: 20}}> Enter the attendance code </Text>
          <TextInput
            style={{marginTop: 5, backgroundColor: '#fff', padding: 5, width: '80%', height: '7%', fontSize: 25, alignSelf: 'center', height: 40}}
            onChangeText={(text) => this.setState({uriPath:text, shouldRenderWebView: false})}
            autoCorrect={false}
            autoCapitalize={'none'}
            value={this.state.uriPath}
            onEndEditing={() => this.renderWebView()}
          />
          <Button
            onPress={() => this.renderWebView()}
            title="Click here to sign in."
            color="#0D6275"
            style={{flex: 1}}
            accessibilityLabel="Attendance Button"
          />
          {/* Either renders instructions or the Attendance website depending on the state */}
          {!this.state.shouldRenderWebView && <View style={{flex:1}}>
              <Image style={{flex:1}} source={{uri:'https://i.imgur.com/szmHnsh.png'}} />
            </View>}
          {this.state.shouldRenderWebView && <WebView
            source={{uri:'https://tinyurl.com/' + this.state.uriPath}}
            style={{marginTop:5, flex:1}}
            renderLoading={this.renderLoadingView} startInLoadingState={true}
          /> }
        </View>
      </SafeAreaView>
    );
  }
}