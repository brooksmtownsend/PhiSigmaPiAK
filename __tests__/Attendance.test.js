var React = require('react');
var ReactNative = require('react-native');
var renderer = require('react-test-renderer');
var jest = require('jest');

import Attendance from '../app/screens/Attendance'

describe("Attendance Functionality", () => {
  it("renders correctly", () => {
    const attendance = renderer.create(<Attendance />).toJSON()
    expect(attendance).toMatchSnapshot()
  })
})
