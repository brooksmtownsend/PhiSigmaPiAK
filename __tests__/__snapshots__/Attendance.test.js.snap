// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attendance Functionality renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 20,
        }
      }
    >
       Enter the attendance code 
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "fontSize": 25,
          "height": 40,
          "marginTop": 5,
          "padding": 5,
          "width": "80%",
        }
      }
      value=""
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Attendance Button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#0D6275",
              },
            ]
          }
        >
          Click here to sign in.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://i.imgur.com/szmHnsh.png",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
